http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/ChannelPipelineInitializer.java

```diff
@@ -167,11 +183,43 @@ private void configureHttp11(Channel ch, ChannelPipeline pipeline) {
         ch.attr(PROTOCOL_FUTURE).get().complete(Protocol.HTTP1_1);
     }
 
+    private void configureAlpnH2(ChannelPipeline pipeline) {
+        pipeline.addLast(new ApplicationProtocolNegotiationHandler("") {
+            @Override
+            protected void configurePipeline(ChannelHandlerContext ctx, String protocol) {
+                if (protocol.equals(ApplicationProtocolNames.HTTP_2)) {
+                    configureHttp2(ctx.channel(), ctx.pipeline());
+                } else {
+                    throw new UnsupportedOperationException("The server does not support ALPN with H2");
+                }
```

Comment:
We should
 - call `ctx.fireExceptionCaught()` here to propagate the exception down the pipeline
 - call `ctx.close()` to close the channel